code/game/machinery/doors/windowdoor.dm

```diff
@@ -317,6 +317,14 @@
 /obj/machinery/door/window/narsie_act()
 	add_atom_colour(NARSIE_WINDOW_COLOUR, FIXED_COLOUR_PRIORITY)
 
+/obj/machinery/door/window/rust_heretic_act()
+	add_atom_colour("#917c65", FIXED_COLOUR_PRIORITY)
+	armor = null
+	max_integrity *= 0.5
+	atom_integrity *= 0.5
```

Comment:
I don't think you're supposed to be setting this directly
`set_armor(/datum/armor/none`, `take_damage(max_integrity * 0.5)`, `modify_max_integrity(max_integrity * 0.5)`
(the order between taking damage and updating max here matters since updating max scales integrity percentage-wise)